app/gateways/http/authentication/middleware.go

```diff
@@ -0,0 +1,43 @@
+package authentication
+
+import (
+	"context"
+	http_server "github.com/patriciapedrosaa/transfer-me/app/gateways/http"
+	"net/http"
+	"strings"
+)
+
+var (
+	ErrEmptyHeader       = "empty authorization header"
+	ErrEmptyToken        = "empty token"
+	ErrInvalidAuthMethod = "invalid auth method"
+	ErrInvalidToken      = "invalid token"
+)
+
+func (h Handler) Authenticate(next http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		authHeader := r.Header.Get("Authorization")
+		if authHeader == "" {
+			http_server.ResponseError(w, http.StatusUnauthorized, ErrEmptyHeader)
+			return
+		}
+		tokenString := strings.Split(authHeader, " ")
+		if len(tokenString) != 2 {
+			http_server.ResponseError(w, http.StatusUnauthorized, ErrEmptyToken)
+			return
+		}
+		if tokenString[0] != "Bearer" {
+			http_server.ResponseError(w, http.StatusUnauthorized, ErrInvalidAuthMethod)
+			return
+		}
```

Comment:
those cases fit better with bad request, since it's an issue with the fulfillment of Authorization header